docs/how-to-build-with-ddn/with-openapi.mdx

```diff
@@ -0,0 +1,261 @@
+```
+
+:::note
+
+The OpenAPI Connector internally makes use of the TypeScript Lambda Connector to expose the generated code as a GraphQL
+API. Read more about the TypeScript Lambda Connector [here](../business-logic/add-a-lambda-connector.mdx).
+
+:::
+
+### Step 5. Add your API
+
+```sh title="Now, track an API from your OpenAPI Document as a command in your DDN metadata:"
+ddn commands add my_openapi getStoreGetOrderById
+```
+
+Open the `app/metadata` directory and you'll find a newly-generated file: `GetStoreGetOrderById.hml`. The DDN CLI will
+use this Hasura Metadata Language file to represent the `getStoreGetOrderById` function from `functions.ts` file as
+[command](/reference/metadata-reference/commands.mdx). This function in turn represents the
+`GET :/store/order/{orderId}` API in your OpenAPI Document.
+
+:::tip
+
+You can import all APIs from your OpenAPI Document as commands by running `ddn command add my_openapi "*"`.
+
+:::
+
+### Step 6. Create a new build
+
+```sh title="To create a local build, run:"
+ddn supergraph build local
+```
+
+The build is stored as a set of JSON files in `engine/build`.
+
+### Step 7. Start your local services
+
+```sh title="Start your local Hasura DDN Engine and OpenAPI connector:"
+ddn run docker-start
+```
+
+Your terminal will be taken over by logs for the different services.
+
+### Step 8. Run your first query
+
+```sh title="In a new terminal tab, open your local console:"
+ddn console --local
+```
+
+```graphql title="In the GraphiQL explorer of the console, write this query:"
+query MyQuery {
+  getStoreGetOrderById(orderId: 10) {
+    complete
+    id
+    petId
+    quantity
+    shipDate
+    status
+  }
+}
+```
+
+```json title="You'll get a similar response to the following:"
+{
+  "data": {
+    "getStoreGetOrderById": {
+      "complete": true,
+      "id": 10,
+      "petId": 198772,
+      "quantity": 7,
+      "shipDate": "2025-03-04T08:23:30.139+00:00",
+      "status": "approved"
+    }
+  }
+}
+```
+
+:::note
+
+The Petstore API is a public API and may have different data than what is shown here. Also, sometimes the API may not
+return data for certain IDs, or, the server may not be accepting requests at all. These will show as errors in the
+console.
+
+:::
+
+### Step 9. Iterate on your OpenAPI Document
+
+If you have added new APIs to your OpenAPI Document, you can introspect it again to update your metadata. The OpenAPI
+connector looks at the `NDC_OAS_FILE_OVERWRITE` to determine whether it should overwrite the existing `api.ts` and
+`functions.ts` file or not. You'll need to set this to `true` in your project's `.env` file if you wish to get the new
+changes from your OpenAPI Document into the generated TypeScript files.
+
+#### Step 9.1 Set the NDC_OAS_FILE_OVERWRITE env var to true
+
+In your projects `.env` file, set the `APP_MY_OPENAPI_NDC_OAS_FILE_OVERWRITE` to `true`.
+
+### Step 11. Refresh your metadata and rebuild your project
```

Comment:
Missing Step 10

Reply:
resolved in [b43294c](https://github.com/hasura/ddn-docs/pull/928/commits/b43294cb58c71b7ebfca24c06e64cfae05f8d888)